alacritty/src/display.rs

```diff
@@ -474,12 +474,18 @@ impl Display {
             None
         };
 
+        let search_regex = search_state.regex.as_ref().map(|regex| Self::format_search(&regex));
+        let search_label = match search_state.direction {
+            Direction::Right => FORWARD_SEARCH_LABEL,
+            Direction::Left => BACKWARD_SEARCH_LABEL,
+        };
+
         // Update IME position.
         #[cfg(not(windows))]
         {
             let point = match &search_regex {
                 Some(regex) => {
-                    let column = min(regex.len() + SEARCH_LABEL.len() - 1, terminal.cols().0 - 1);
+                    let column = min(regex.len() + search_label.len() - 1, terminal.cols().0 - 1);
                     Point::new(terminal.screen_lines() - 1, Column(column))
                 },
```

Comment:
I'd like to move this to a different location maybe. It would be possible to just store the grid's cursor point, drop the terminal and then update the IME at a later point. Since the cursor is cheap to copy that shouldn't be a problem.

That way we could then maybe update the IME in a branch together with search, so we wouldn't have to check that search is enabled in draw_search but would just pass the desugared search `String` (or `&str`) without `Option` directly to `draw_search`. This could also make it possible to maybe already get rid of the search label, but that might be tough.